keps/sig-node/2403-pod-resources-allocatable-resources/README.md

```diff
@@ -207,67 +242,144 @@ To a vendor changes in the API should always be backwards compatible.
-* **Were upgrade and rollback tested? Was upgrade->downgrade->upgrade path tested?** Not Applicable.
-* **Is the rollout accompanied by any deprecations and/or removals of features,  APIs, fields of API types, flags, etc.?** No.
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Kubelet may fail to start. The new API may report inconsistent data, or may cause the kubelet to crash.
+
+###### What specific metrics should inform a rollback?
+
+`pod_resources_endpoint_errors_get_allocatable` - but only with feature gate enabled. Otherwise the API will always return a known error, giving a false negative signal.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Not Applicable.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No (Not applicable)
 
-### Monitoring requirements
-* **How can an operator determine if the feature is in use by workloads?**
-  - Look at the `pod_resources_endpoint_requests_get_allocatable` metric exposed by the kubelet.
-  - Clients are connected to the podresources unix socket, for example  bychecking which containers mount the podresources socket path.
-* **What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?**
-  - [X] Metrics
-    - Metric name: `pod_resources_endpoint_requests_total`, `pod_resources_endpoint_requests_list`, `pod_resources_endpoint_requests_get_allocatable`, `pod_resources_endpoint_errors_list`, `pod_resources_endpoint_errors_get_allocatable`
-    - Components exposing the metric: kubelet
+### Monitoring Requirements
 
-* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?** N/A.
-* **Are there any missing metrics that would be useful to have to improve observability if this feature?** As part of this feature enhancement, per-API-endpoint resources metrics are being added; to observe this feature the `pod_resources_endpoint_requests_get_allocatable` metric should be used. We will also add error counting metrics to improve the observability of the API.
+###### How can an operator determine if the feature is in use by workloads?
 
+- Look at the `pod_resources_endpoint_requests_get_allocatable` metric exposed by the kubelet.
+- Clients are connected to the podresources unix socket, for example by checking which containers mount the podresources socket path.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [ ] Events
+  - Event Reason: 
+- [ ] API .status
+  - Condition name: 
+  - Other field: 
+- [X] Other (treat as last resort)
+  - Look at the `pod_resources_endpoint_requests_get_allocatable` and `pod_resources_endpoint_errors_get_allocatable` metrics exposed by the kubelet.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+Not Applicable
```

Comment:
The section below this mentions adding error counting metrics and the diff from the previous version mentioned a `pod_resources_endpoint_errors_list` metric, along with a `pod_resources_endpoint_requests_total` and `pod_resources_endpoint_requests_list` metric that seem to be missing from here. Does this have an expected 100% availability if the kubelet is running?

Reply:
yes, because this API ultimately exposes, in read only mode, (a processed form of) kubelet internal data which is critical for functioning of the kubelet.

I'll make this information more explicit in the KEP text.